src/cmd/vm.rs

```diff
@@ -20,136 +27,231 @@ pub struct Args {
         SubCommand::Setup(args) => run_setup(args),
         SubCommand::Start(args) => run_start(args),
-        SubCommand::Push(args) => run_push(args),
     }
 }
 
 #[derive(FromArgs, PartialEq, Debug)]
-/// connects to a running betty instance via SSH
-#[argh(subcommand, name = "connect")]
-pub struct ArgsConnect {
-    /// extra arguments
+/// run first time setup, installs necessary dependencies
+#[argh(subcommand, name = "setup")]
+pub struct ArgsSetup {
+    /// path to dir where betty.sh exists. If omitted, current directory will be
+    /// used.
+    #[argh(option)]
+    arc: Option<String>,
+
+    /// extra arguments to pass to betty.sh. You can pass other options like
+    /// --extra-args "options".
     #[argh(option)]
-    extra_args: Vec<String>,
+    extra_args: Option<String>,
 }
 
-fn run_connect(_args: &ArgsConnect) -> Result<()> {
+fn run_setup(args: &ArgsSetup) -> Result<()> {
+    let dir = find_betty_script(&args.arc)?;
+
+    println!("Updating packages...");
+    let update_package = Command::new("sudo")
+        .args(["apt", "update"])
+        .spawn()
+        .context("Failed to execute sudo apt update")?;
+    update_package
+        .wait_with_output()
+        .context("Failed to wait for updating packages")?;
+
+    println!("Enable KVM...");
+    enable_kvm()?;
+
+    println!("Installing python packages...");
+    let install_python_package = Command::new("sudo")
+        .args(["apt", "install", "python3-pip", "python3-venv"])
+        .spawn()
+        .context("Failed to install python packages")?;
+    install_python_package
+        .wait_with_output()
+        .context("Failed to wait for installing python packages")?;
+
+    println!("Running betty.sh setup...");
+    let arg = match &args.extra_args {
+        Some(a) => a,
+        None => "",
+    };
+    run_betty(&dir, "setup", arg)?;
+
+    println!("Running gcloud auth login...");
+    let gcloud_auth = Command::new("gcloud")
+        .args(["auth", "login"])
+        .spawn()
+        .context("Failed to run gcloud login gcloud")?;
+    gcloud_auth
+        .wait_with_output()
+        .context("Failed to wait for gcloud auth login")?;
+
     Ok(())
 }
 
-#[derive(FromArgs, PartialEq, Debug)]
-/// run first time setup, installs necessary dependencies
-#[argh(subcommand, name = "setup")]
-pub struct ArgsSetup {
-    /// extra arguments
-    #[argh(option)]
-    extra_args: Vec<String>,
-}
+fn enable_kvm() -> Result<()> {
+    let username = whoami::username();
+
+    println!("Installing kvm support...");
+    let install_kvm_support = Command::new("sudo")
+        .args(["apt-get", "install", "qemu-system-x86"])
+        .spawn()
+        .context("Failed to install kvm support")?;
+    install_kvm_support
+        .wait_with_output()
+        .context("Failed to wait for installing kvm support")?;
+
+    println!("Loading Kernel modules...");
+    let load_kernel_module = Command::new("sudo")
+        .args(["modprobe", "kvm-intel"])
```

Comment:
is it possible to be running on an AMD machine for a cloudtop? I guess not for now

Reply:
Thank you for letting me know, I added the support for AMD machines.  
f5908dbe199e6460706597c194fe4c44fb798e9d